### ISSUE TITLE
unable to compile on windows

### ISSUE THREAD

**Issue by dpastoor** [r-lib/later#6]
Hi Joe,

Any ideas as to the problem with the following compilation error

```
> devtools::install_github("r-lib/later")
Using GitHub PAT from envvar GITHUB_PAT
Downloading GitHub repo r-lib/later@master
from URL https://api.github.com/repos/r-lib/later/zipball/master
Installing later
"C:/PROGRA~1/R/R-34~1.0/bin/x64/R" --no-site-file --no-environ --no-save --no-restore --quiet CMD INSTALL  \
  "C:/Users/devin/AppData/Local/Temp/RtmpoXC0Bm/devtools636c246f1b26/r-lib-later-8d6b106" --library="C:/Users/devin/Documents/R/win-library/3.4" --install-tests 

* installing *source* package 'later' ...
** libs

*** arch - i386
c:/Rtools/mingw_32/bin/g++  -I"C:/PROGRA~1/R/R-34~1.0/include" -DNDEBUG -pthread -I"C:/Users/devin/Documents/R/win-library/3.4/Rcpp/include" -I"C:/Users/devin/Documents/R/win-library/3.4/BH/include"   -I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -mtune=core2 -c RcppExports.cpp -o RcppExports.o
c:/Rtools/mingw_32/bin/g++  -I"C:/PROGRA~1/R/R-34~1.0/include" -DNDEBUG -pthread -I"C:/Users/devin/Documents/R/win-library/3.4/Rcpp/include" -I"C:/Users/devin/Documents/R/win-library/3.4/BH/include"   -I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -mtune=core2 -c callback_registry.cpp -o callback_registry.o
c:/Rtools/mingw_32/bin/gcc  -I"C:/PROGRA~1/R/R-34~1.0/include" -DNDEBUG -pthread -I"C:/Users/devin/Documents/R/win-library/3.4/Rcpp/include" -I"C:/Users/devin/Documents/R/win-library/3.4/BH/include"   -I"d:/Compiler/gcc-4.9.3/local330/include"     -O3 -Wall  -std=gnu99 -mtune=core2 -c init.c -o init.o
c:/Rtools/mingw_32/bin/g++  -I"C:/PROGRA~1/R/R-34~1.0/include" -DNDEBUG -pthread -I"C:/Users/devin/Documents/R/win-library/3.4/Rcpp/include" -I"C:/Users/devin/Documents/R/win-library/3.4/BH/include"   -I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -mtune=core2 -c later.cpp -o later.o
c:/Rtools/mingw_32/bin/g++  -I"C:/PROGRA~1/R/R-34~1.0/include" -DNDEBUG -pthread -I"C:/Users/devin/Documents/R/win-library/3.4/Rcpp/include" -I"C:/Users/devin/Documents/R/win-library/3.4/BH/include"   -I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -mtune=core2 -c later_native.cpp -o later_native.o
c:/Rtools/mingw_32/bin/g++  -I"C:/PROGRA~1/R/R-34~1.0/include" -DNDEBUG -pthread -I"C:/Users/devin/Documents/R/win-library/3.4/Rcpp/include" -I"C:/Users/devin/Documents/R/win-library/3.4/BH/include"   -I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -mtune=core2 -c later_posix.cpp -o later_posix.o
c:/Rtools/mingw_32/bin/g++  -I"C:/PROGRA~1/R/R-34~1.0/include" -DNDEBUG -pthread -I"C:/Users/devin/Documents/R/win-library/3.4/Rcpp/include" -I"C:/Users/devin/Documents/R/win-library/3.4/BH/include"   -I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -mtune=core2 -c later_win32.cpp -o later_win32.o
c:/Rtools/mingw_32/bin/g++  -I"C:/PROGRA~1/R/R-34~1.0/include" -DNDEBUG -pthread -I"C:/Users/devin/Documents/R/win-library/3.4/Rcpp/include" -I"C:/Users/devin/Documents/R/win-library/3.4/BH/include"   -I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -mtune=core2 -c timer_posix.cpp -o timer_posix.o
c:/Rtools/mingw_32/bin/g++  -I"C:/PROGRA~1/R/R-34~1.0/include" -DNDEBUG -pthread -I"C:/Users/devin/Documents/R/win-library/3.4/Rcpp/include" -I"C:/Users/devin/Documents/R/win-library/3.4/BH/include"   -I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -mtune=core2 -c timestamp_unix.cpp -o timestamp_unix.o
c:/Rtools/mingw_32/bin/g++  -I"C:/PROGRA~1/R/R-34~1.0/include" -DNDEBUG -pthread -I"C:/Users/devin/Documents/R/win-library/3.4/Rcpp/include" -I"C:/Users/devin/Documents/R/win-library/3.4/BH/include"   -I"d:/Compiler/gcc-4.9.3/local330/include"     -O2 -Wall  -mtune=core2 -c timestamp_win32.cpp -o timestamp_win32.o
timestamp_win32.cpp: In constructor 'Timestamp::Timestamp()':
timestamp_win32.cpp:40:56: error: invalid new-expression of abstract class type 'TimestampImplWin32'
 Timestamp::Timestamp() : p_impl(new TimestampImplWin32()) {}
                                                        ^
timestamp_win32.cpp:7:7: note:   because the following virtual functions are pure within 'TimestampImplWin32':
 class TimestampImplWin32 : public TimestampImpl {
       ^
In file included from timestamp_win32.cpp:3:0:
timestamp.h:13:18: note: 	virtual double TimestampImpl::diff_secs(const TimestampImpl*) const
   virtual double diff_secs(const TimestampImpl* other) const = 0;
                  ^
timestamp_win32.cpp: In constructor 'Timestamp::Timestamp(double)':
timestamp_win32.cpp:41:71: error: invalid new-expression of abstract class type 'TimestampImplWin32'
 Timestamp::Timestamp(double secs) : p_impl(new TimestampImplWin32(secs)) {}
                                                                       ^
make: *** [timestamp_win32.o] Error 1
Warning: running command 'make -f "Makevars" -f "C:/PROGRA~1/R/R-34~1.0/etc/i386/Makeconf" -f "C:/PROGRA~1/R/R-34~1.0/share/make/winshlib.mk" SHLIB_LDFLAGS='$(SHLIB_CXXLDFLAGS)' SHLIB_LD='$(SHLIB_CXXLD)' SHLIB="later.dll" OBJECTS="RcppExports.o callback_registry.o init.o later.o later_native.o later_posix.o later_win32.o timer_posix.o timestamp_unix.o timestamp_win32.o"' had status 2
ERROR: compilation failed for package 'later'
```

cherry-picked package info if you need it:

```
> devtools::session_info()
Session info -----------------------------------------------------------------------------------------------------------------------------------------------------
 setting  value                       
 version  R version 3.4.0 (2017-04-21)
 system   x86_64, mingw32             
 ui       RStudio (1.1.327)           
 language (EN)                        
 collate  English_United States.1252  
 tz       America/New_York            
 date     2017-08-18             

 Rcpp        0.12.12 2017-07-15 CRAN (R 3.4.1)
 BH        1.62.0-1 2016-11-19 CRAN (R 3.4.0)
     
```

**Comment by dpastoor** [comment]
Tracking this along with the issue that prompted trying to explore the dev version https://github.com/thomasp85/fiery/issues/15

**Comment by jcheng5** [comment]
That error seems to indicate that `_WIN32` isn't defined. Are you using cygwin or something?

**Comment by dpastoor** [comment]
using the default Rtools toolchain

**Comment by jcheng5** [comment]
I broke it, argh. Forgot to implement `diff_secs` apparently.

**Comment by jcheng5** [comment]
@dpastoor Try now?

**Comment by dpastoor** [comment]
great we're in business! Fixed in 1d15d8a9d95a96e8c0753af2634321913773892f

**Comment by jcheng5** [comment]
Great, thanks. Sorry for the inconvenience!
